pkg/scheduler/factory.go

```diff
@@ -113,181 +109,104 @@ type Config struct {
+
+	nodeInfoSnapshot *nodeinfosnapshot.Snapshot
 
 	factoryArgs        *PluginFactoryArgs
 	configProducerArgs *plugins.ConfigProducerArgs
-}
 
-// ConfigFactoryArgs is a set arguments passed to NewConfigFactory.
-type ConfigFactoryArgs struct {
-	Client                         clientset.Interface
-	InformerFactory                informers.SharedInformerFactory
-	NodeInformer                   coreinformers.NodeInformer
-	PodInformer                    coreinformers.PodInformer
-	PvInformer                     coreinformers.PersistentVolumeInformer
-	PvcInformer                    coreinformers.PersistentVolumeClaimInformer
-	ReplicationControllerInformer  coreinformers.ReplicationControllerInformer
-	ReplicaSetInformer             appsinformers.ReplicaSetInformer
-	StatefulSetInformer            appsinformers.StatefulSetInformer
-	ServiceInformer                coreinformers.ServiceInformer
-	PdbInformer                    policyinformers.PodDisruptionBudgetInformer
-	StorageClassInformer           storageinformersv1.StorageClassInformer
-	CSINodeInformer                storageinformersv1beta1.CSINodeInformer
-	VolumeBinder                   *volumebinder.VolumeBinder
-	SchedulerCache                 internalcache.Cache
-	HardPodAffinitySymmetricWeight int32
-	DisablePreemption              bool
-	PercentageOfNodesToScore       int32
-	BindTimeoutSeconds             int64
-	PodInitialBackoffSeconds       int64
-	PodMaxBackoffSeconds           int64
-	StopCh                         <-chan struct{}
-	Registry                       framework.Registry
-	Plugins                        *schedulerapi.Plugins
-	PluginConfig                   []schedulerapi.PluginConfig
-	PluginConfigProducerRegistry   *plugins.ConfigProducerRegistry
-}
-
-// NewConfigFactory initializes the default implementation of a Configurator. To encourage eventual privatization of the struct type, we only
-// return the interface.
-func NewConfigFactory(args *ConfigFactoryArgs) *Configurator {
-	stopEverything := args.StopCh
-	if stopEverything == nil {
-		stopEverything = wait.NeverStop
-	}
-
-	// storageClassInformer is only enabled through VolumeScheduling feature gate
-	var storageClassLister storagelistersv1.StorageClassLister
-	if args.StorageClassInformer != nil {
-		storageClassLister = args.StorageClassInformer.Lister()
-	}
+	// Always check all predicates even if the middle of one predicate fails.
+	alwaysCheckAllPredicates bool
 
-	var csiNodeLister storagelistersv1beta1.CSINodeLister
-	if args.CSINodeInformer != nil {
-		csiNodeLister = args.CSINodeInformer.Lister()
-	}
+	enableNonPreempting bool
+}
 
-	var pdbLister policylisters.PodDisruptionBudgetLister
-	if args.PdbInformer != nil {
-		pdbLister = args.PdbInformer.Lister()
+// complete completes configurator object's creation by filling fields with default values.
+// It should be called before any call to Configurator.Create*
+func (c *Configurator) complete() {
```

Comment:
everything in this function can be set directly, I don't think we need the function ```complete```

Reply:
You mean wherever they are used? If we can get rid of all of these unexported fields easily now, let's do it. Otherwise I'm happy to leave it as a follow up.

Reply:
They are used in one place, aren't they? 

I feel we need a couple of cleanup PRs before this one, one to replace all listers with SharedInformerFactory, and the other to use SharedLister instead of the individual listers it represents. Let me try to do that so we can get this done before the freeze.

Reply:
There you go, I think we should merge https://github.com/kubernetes/kubernetes/pull/85150 first before this one.